### ISSUE TITLE
Improvement: Center align util method

### ISSUE THREAD

**Issue by bamdadfr** [nvimdev/dashboard-nvim#402]
Hi,

### TLDR

I changed the behaviour of `utils.center_align` as its current behaviour does
not fit my use case. The injected string is shifted line by line rather than
indenting the whole block. I don't know if I should either:

- open a PR, and overwrite the align function
- open a PR, and add a new align function + user setting
- keep it to my fork

### Issue

I was playing with the multi line string rendered in the dashboard in order to
display custom ASCII art. Unfortunately, lines with numerous spaces around
are not properly rendered.

After reading `utils.center_align` method, I noticed that alignment is done on
a line basis instead of considering the whole block's indentation.

Please find below the code change and the different renderings.

### Code change

In [`utils.lua:53`](https://github.com/nvimdev/dashboard-nvim/blob/63df28409d940f9cac0a925df09d3dc369db9841/lua/dashboard/utils.lua#L53)

#### Before

The current code applies shifts to each line independently.

```lua
local centered_lines = {}
local fills = fill_sizes(tbl)

for i = 1, #tbl do
  local fill_line = (' '):rep(fills[i]) .. tbl[i]
  table.insert(centered_lines, fill_line)
end
```

#### After

We add a `fill_min` variable to indent all lines by the same amount.

```lua
local centered_lines = {}
local fills = fill_sizes(tbl)
local fill_min = math.min(unpack(fills))

for i = 1, #tbl do
  local fill_line = (' '):rep(fill_min) .. tbl[i]
  table.insert(centered_lines, fill_line)
end
```

Please find below this wonderful spider I downloaded
[here](http://www.ascii-art.de/ascii/s/spider.txt).

### Raw string (input)

```text
                   /\
                  /  \
                 |  _ \                  _
                 | / \ \                / \
                 |/   \ \              /   \
                 /     \ |        /\  /     \
                /|      \| ~  ~  /  \/       \
        _______/_|_______\(o)(o)/___/\_____   \
       /      /  |       (______)     \    \   \_
      /      /   |                     \    \
     /      /    |                      \    \
    /      /     |                       \    \
   /     _/      |                        \    \
  /             _|                         \    \_
_/                                          \
                                             \
                                              \_
                                             ~tahl~
```

### Neovim render (before)

![before](https://github.com/nvimdev/dashboard-nvim/assets/17913954/619be807-d79f-46e0-ae3d-8c510482d4ca)

### Neovim render (after)

![after](https://github.com/nvimdev/dashboard-nvim/assets/17913954/ce75fbe5-a761-4cc3-bc99-2f73d510f6b5)
